src/instructlab/lab.py

```diff
@@ -1064,40 +1078,52 @@ def train(
 
         # TODO: Figure out what to do when there are multiple checkpoint dirs.
         # Right now it's just copying files from the first one numerically not necessarily the best one
-        added_tokens_file = glob(
-            training_results_dir + "/checkpoint-*/added_tokens.json"
-        )
-        special_tokens_map = glob(
-            training_results_dir + "/checkpoint-*/special_tokens_map.json"
-        )
-        tokenizer_json = glob(training_results_dir + "/checkpoint-*/tokenizer.json")
-        tokenizer_model = glob(training_results_dir + "/checkpoint-*/tokenizer.model")
-        tokenizer_config_json = glob(
-            training_results_dir + "/checkpoint-*/tokenizer_config.json"
-        )
+        added_tokens_file = os.path.join(best_checkpoint, "added_tokens.json")
+        special_tokens_map = os.path.join(best_checkpoint, "special_tokens_map.json")
+        tokenizer_json = os.path.join(best_checkpoint, "tokenizer.json")
+        tokenizer_model = os.path.join(best_checkpoint, "tokenizer.model")
+        tokenizer_config_json = os.path.join(best_checkpoint, "tokenizer_config.json")
         config_json = glob(training_results_dir + "/merged_model/config.json")
         generation_config_json = glob(
             training_results_dir + "/merged_model/generation_config.json"
         )
         safe_tensors = glob(training_results_dir + "/merged_model/*.safetensors")
 
-        shutil.copy(added_tokens_file[0], final_results_dir)
-        print("Copied ", added_tokens_file[0], "to ", final_results_dir)
-        shutil.copy(special_tokens_map[0], final_results_dir)
-        print("Copied ", special_tokens_map[0], "to ", final_results_dir)
-        shutil.copy(tokenizer_json[0], final_results_dir)
-        print("Copied ", tokenizer_json[0], "to ", final_results_dir)
-        shutil.copy(tokenizer_model[0], final_results_dir)
-        print("Copied ", tokenizer_model[0], "to ", final_results_dir)
-        shutil.copy(tokenizer_config_json[0], final_results_dir)
-        print("Copied ", tokenizer_config_json[0], "to ", final_results_dir)
-        shutil.copy(config_json[0], final_results_dir)
-        print("Copied ", config_json[0], "to ", final_results_dir)
-        shutil.copy(generation_config_json[0], final_results_dir)
-        print("Copied ", generation_config_json[0], "to ", final_results_dir)
+        try:
+            shutil.copy(added_tokens_file, final_results_dir)
+            print("Copied ", added_tokens_file, "to ", final_results_dir)
```

Comment:
I know it was there before but if someone has the answer I'll be happy to hear it, why do we print and not user a logger here? Additionally, those look like DEBUG-level information to me.

Reply:
Not sure, @alimaredia any specific reason?

Reply:
I would def be in favor of using the logger for this, but reasonable to do this as a followup and not as part of this PR IMHO